File: backend/app/schemas/tag.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class TagBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    color: str | None = Field(default=None, pattern=r"^#[0-9a-fA-F]{6}$")


class TagCreate(TagBase):
    pass


class TagUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=255)
    color: str | None = Field(default=None, pattern=r"^#[0-9a-fA-F]{6}$")


class TagRead(TagBase):
    id: UUID
    user_id: UUID
    created_at: datetime
    updated_at: datetime


class TagListResponse(BaseModel):
    data: list[TagRead]


class TagId(BaseModel):
    id: UUID


class TagIdResponse(BaseModel):
    data: TagId


class GameTagLinkRequest(BaseModel):
    tag_id: UUID

File: frontend/src/types/tag.ts
export type Tag = {
  id: string;
  user_id: string;
  name: string;
  color: string | null;
  created_at: string;
  updated_at: string;
};

export type TagListResponse = {
  data: Tag[];
};

export type TagIdResponse = {
  data: {
    id: string;
  };
};

export type TagCreatePayload = {
  name: string;
  color?: string | null;
};
